playerctrl/controller.go

```diff
@@ -249,3 +250,31 @@ func (ctrl *Controller) notifyClients(sessionID string, action notifyAction) {
 		action(spotifyClient)
 	}
 }
+
+// sends out a player state change event with relevant data about the current player state
+func (ctrl *Controller) notifyPlayerStateChange(sessionID string) {
+	msg := "[playerctrl] notify player state change"
+	ctx := context.Background()
+
+	playr, err := ctrl.playerCollection.GetPlayer(ctx, sessionID)
+	if err != nil {
+		log.Errorf("%v: %v", msg, err)
+		return
+	}
+
+	playerState := &struct {
+		CurrentSong *song.Model   `json:"current_song"`
+		IsPlaying   bool          `json:"is_playing"`
+		Progress    time.Duration `json:"progress"`
+	}{
+		CurrentSong: playr.CurrentSong,
+		IsPlaying:   !playr.Paused,
+		Progress:    playr.Progress(),
+	}
+
```

Comment:
can you  define a struct `PlayerStateChangePayload` in `sse`
have a look at `playerctrl/events`